common/src/main/java/com/wynntils/models/character/SkillPointModel.java

```diff
@@ -91,22 +88,28 @@ public void saveCurrentSkillPoints(String name) {
      * Saves the current equipped gear and provided skill points.
      */
     public void saveBuild(String name, int[] skillPoints) {
+        String weapon = null;
         List<String> armourNames = new ArrayList<>();
-        McUtils.inventory().armor.stream()
-                .filter(x -> !x.isEmpty())
-                .map(x -> x.getHoverName().getString())
-                .forEach(armourNames::add);
-        Collections.reverse(armourNames); // helmet to boots order
-
         List<String> accessoryNames = new ArrayList<>();
-        for (int i : ACCESSORY_SLOTS) {
-            ItemStack itemStack = McUtils.inventory().getItem(i);
-            if (!itemStack.isEmpty() && !itemStack.getHoverName().getString().equals(EMPTY_ACCESSORY_SLOT)) {
-                accessoryNames.add(itemStack.getHoverName().getString());
+
+        for (ItemStack itemStack : Models.PlayerInventory.getEquippedItems()) {
+            Optional<WynnItem> wynnItemOptional = Models.Item.getWynnItem(itemStack);
+            if (wynnItemOptional.isEmpty()) continue;
+            WynnItem wynnItem = wynnItemOptional.get();
+
+            if (wynnItem instanceof GearTypeItemProperty gear) {
+                if (gear.getGearType().isArmor()) {
+                    armourNames.add(itemStack.getHoverName().getString());
+                } else if (gear.getGearType().isAccessory()) {
+                    accessoryNames.add(itemStack.getHoverName().getString());
+                } else if (gear.getGearType().isWeapon()) {
+                    weapon = itemStack.getHoverName().getString();
```

Comment:
Usually we don't use Component#getString, because it can produce unexpected output when Wynn changes some small thing in their back-end. Use StyledText#getStringWithoutFormatting